### DISCUSSION TITLE
Easiest way to pull in a Rust repository that produces a binary as an external dependency

### DISCUSSION THREAD

**Discussion by mikn** [bazelbuild/rules_rust#3121] [Q&A]
We have a monorepo that has implemented Bzlmod with a few languages in it, but we do not yet have Rust - we have some tools in our toolchain however that is written in Rust (including our own tooling in separate repositories). These projects are however not always published crates (and for our own tooling, we want to support a local development flow), and they are not Bzlmod repositories.
What would be the easiest way for us to "import" these projects, have rules_rust build them (including all of their external crate dependencies) and expose the binary that is produced?

If they were published crates, it would be easy to use this pattern:
```
crate.spec(package = "protoc-gen-tonic", version = "0.4.0")
crate.annotation(crate = "protoc-gen-tonic", gen_binaries = ["protoc-gen-tonic"])
```

But since the binaries are not published crates, I am unsure how to go about this.

**Comment by illicitonion** [comment]
What _are_ they? Are they directories with `Cargo.toml` files that exist in some git repository somewhere? If so, you could use a `git` dependency in a `Cargo.toml` file used for the `crate_repository`.
